SMLHelper/Patchers/CraftDataPatcher_Subnautica.cs

```diff
@@ -1,97 +1,301 @@
+        {
+            if (CustomCookedCreatureList.Count > 0 && CookedCreatureAltered < CustomCookedCreatureList.Count)
+            {
+                PatchUtils.PatchDictionary(CraftData.cookedCreatureList, CustomCookedCreatureList);
+                CookedCreatureAltered = CustomCookedCreatureList.Count;
+            }
+        }
+
+        private static void GetBackgroundTypesPrefix()
+        {
+            if (CustomBackgroundTypes.Count > 0 && BackgroundTypesAltered < CustomBackgroundTypes.Count)
+            {
+                PatchUtils.PatchDictionary(CraftData.backgroundTypes, CustomBackgroundTypes);
+                BackgroundTypesAltered = CustomBackgroundTypes.Count;
+            }
+        }
+
+        private static void GetBuildablePrefix()
+        {
+            if (CustomBuildables.Count > 0 && BuildablesAltered < CustomBuildables.Count)
+            {
+                PatchUtils.PatchList(CraftData.buildables, CustomBuildables);
+                BuildablesAltered = CustomBuildables.Count;
+            }
+        }
+
+        private static void AddCustomDataToOriginalDictionaries()
         {
             short added = 0;
             short replaced = 0;
             foreach (TechType techType in CustomTechData.Keys)
             {
                 ITechData smlTechData = CustomTechData[techType];
+                bool techExists = CraftData.techData.TryGetValue(techType, out CraftData.TechData techData);
+                bool sameData = false;
 
-                var techDataInstance = new CraftData.TechData
+                if (techExists)
                 {
-                    _techType = techType,
-                    _craftAmount = smlTechData.craftAmount
-                };
+                    sameData = smlTechData.craftAmount == techData.craftAmount &&
+                        smlTechData.ingredientCount == techData.ingredientCount &&
+                        smlTechData.linkedItemCount == techData.linkedItemCount;
+                    
+                    if(sameData)
+                        for (int i = 0; i < smlTechData.ingredientCount; i++)
+                        {
+                            if (smlTechData.GetIngredient(i).techType != techData.GetIngredient(i).techType)
+                            {
+                                sameData = false;
+                                break;
+                            }
+                            if (smlTechData.GetIngredient(i).amount != techData.GetIngredient(i).amount)
+                            {
+                                sameData = false;
+                                break;
+                            }
+                        }
 
-                var ingredientsList = new CraftData.Ingredients();
+                    if (sameData)
+                        for (int i = 0; i < smlTechData.linkedItemCount; i++)
+                        {
+                            if (smlTechData.GetLinkedItem(i) != techData.GetLinkedItem(i))
```

Comment:
This may be dangerous if the number of linked items in `smlTechData` is larger than the ones in `techData`.